src/modules/hip/kernel/slice.hpp

```diff
@@ -32,79 +87,487 @@ __global__ void slice_ncdhw_hip_tensor(T *srcPtr,
+            Rpp32u maxHeight = std::min(shape[0], length[0] - anchor[0]);
+            Rpp32u maxWidth = std::min(shape[1], length[1] - anchor[1]);
+
+            // check if padding is needed
+            bool needPadding = (((anchor[0] + shape[0]) > length[0]) ||
+                                ((anchor[1] + shape[1]) > length[1]));
+
+            // launch kernel for filling the padded region with fill value specified
+            if(needPadding)
+            {
+                hipLaunchKernelGGL(fill_value_ncdhw_hip_tensor,
+                                    dim3(ceil((float)globalThreads_x/LOCAL_THREADS_X), ceil((float)globalThreads_y/LOCAL_THREADS_Y), ceil((float)globalThreads_z/LOCAL_THREADS_Z)),
+                                    dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, 1),
+                                    0,
+                                    handle.GetStream(),
+                                    dstPtr + (batchCount * dstGenericDescPtr->strides[0]),
+                                    make_uint3(0, 0, dstGenericDescPtr->strides[1]),
+                                    1,
+                                    make_uint3(1, shape[0], shape[1]),
+                                    fillValue);
+            }
+        }
+    }
+    else if(numDims == 1)
+    {
+        int globalThreads_x = (dstGenericDescPtr->strides[0] + 7) >> 3; // W - width (x direction) - vectorized for 8 element loads/stores per channel
+        int globalThreads_y = 1;
+        int globalThreads_z = 1;
+
+        for(int batchCount = 0; batchCount < dstGenericDescPtr->dims[0]; batchCount++)
+        {
+            Rpp32s *anchor = &anchorTensor[batchCount * numDims];
+            Rpp32s *shape = &shapeTensor[batchCount * numDims];
+            Rpp32u *roi = roiTensor + batchCount * numDims * 2;
+            Rpp32s *length = reinterpret_cast<Rpp32s *>(&roi[numDims]);
+            Rpp32u maxLength = std::min(shape[0], length[0] - anchor[0]);
+
+            // check if padding is needed
+            bool needPadding = ((anchor[0] + shape[0]) > length[0]);
+
+            // launch kernel for filling the padded region with fill value specified
+            if(needPadding)
+            {
+                hipLaunchKernelGGL(fill_value_ncdhw_hip_tensor,
+                                   dim3(ceil((float)globalThreads_x/LOCAL_THREADS_X), ceil((float)globalThreads_y/LOCAL_THREADS_Y), ceil((float)globalThreads_z/LOCAL_THREADS_Z)),
+                                   dim3(LOCAL_THREADS_X, 1, 1),
+                                   0,
+                                   handle.GetStream(),
+                                   dstPtr + (batchCount * dstGenericDescPtr->strides[0]),
+                                   make_uint3(0, 0, 1),
+                                   1,
+                                   make_uint3(1, 1, shape[0]),
+                                   fillValue);
+            }
         }
     }
+    hipStreamSynchronize(handle.GetStream());
```

Comment:
Pls move the stream sync outside. To the line after the call to hip_exec_fill_value_tensor()